test/rational.jl

```diff
@@ -96,6 +96,10 @@ using Test
     @test !(1//3 < NaN)
     @test !(1//3 == NaN)
     @test !(1//3 > NaN)
+
+    @test abs(one(Rational{UInt})) === one(Rational{UInt})
+    @test abs(one(Rational{Int})) === one(Rational{Int})
+    @test abs(-one(Rational{Int})) === one(Rational{Int})
```

Comment:
Would be good to add a comment to reference this PR just above the tests. The two last tests seems unrelated to the PR?

Reply:
You are perfectly right. Just the first test is related to the bug. However, I did not find other tests regarding the abs method, so I thought that checking that the signed version works too would be useful. I admit I have not much experience about writing testcases, so if you think the tests should be kept to a minimum I will remove the last two.

Reply:
These seem fine to me. The main thing is to test it and I don’t mind having a slightly unrelated test added as well.